osquery/tables/system/windows/registry.cpp

```diff
@@ -60,10 +60,32 @@ const std::map<DWORD, std::string> kRegistryTypes = {
     {REG_RESOURCE_LIST, "REG_RESOURCE_LIST"},
 };
 
+const char kRegSep = '\\';
+
+void explodeRegistryPath(const std::string& path,
+                         std::string& rHive, 
+                         std::string& rKey) {
+  size_t sepPos = path.find(kRegSep);
+  if (sepPos != std::string::npos) {
+    rHive = path.substr(0, sepPos);
```

Comment:
Ah, I was digging around for something like that. Probably a bit less efficient but much more concise and readable, and in this case I don't think the tiny efficiency win is worth it.